cmd/kubelet/kubelet.go

```diff
@@ -34,12 +35,15 @@ import (
 )
 
 func main() {
+	defer logs.FlushLogs()
```

Comment:
This should be called before `os.Exit()`, right?

Reply:
Unfortunately no, `os.Exit()` terminates the program but does not ever return from `main()`.  Go does not have an `atexit()` equivalent.

Reply:
Yes. What I meant to say is that this line is not executed when `os.Exit()` is called but it probably should. So `logs.FlushLogs()` should be explicitly (rather than via `defer` which doesn't work) called before calling `os.Exit()` below.

Reply:
That's true. We solved that in `cli.Run` by returning and having `os.Exit` in `main`, but not here. I'll fix it.

Reply:
Grepping for `os.Exit` shows that we have plenty of other places where log flushing gets skipped, for example:
```
cmd/kube-controller-manager/app/controllermanager.go-                   if err := Run(c.Complete(), wait.NeverStop); err != nil {
cmd/kube-controller-manager/app/controllermanager.go-                           fmt.Fprintf(os.Stderr, "%v\n", err)
cmd/kube-controller-manager/app/controllermanager.go:                           os.Exit(1)
cmd/kube-controller-manager/app/controllermanager.go-                   }
```

We probably could have something like `component-base/cli.Exit` that replaces `os.Exit` or perhaps ban `os.Exit` because it skips also all other defers. But let's address that in a separate PR.

Reply:
One more thing I noticed is `CheckErr()` that is used in a lot of places. See https://github.com/kubernetes/kubernetes/issues/102231#issuecomment-930683302

Reply:
So there is even an issue about that. Good, let's continue there.